source/common/http/conn_manager_impl.cc

```diff
@@ -1118,8 +1123,22 @@ void ConnectionManagerImpl::ActiveStream::setBufferLimit(uint32_t new_limit) {
   }
 }
 
-void ConnectionManagerImpl::ActiveStream::createFilterChain() {
+bool ConnectionManagerImpl::ActiveStream::createFilterChain() {
+  bool upgrade_rejected = false;
+  auto upgrade = request_headers_->Upgrade();
+  if (upgrade != nullptr) {
+    if (connection_manager_.config_.filterFactory().createUpgradeFilterChain(
+            upgrade->value().c_str(), *this)) {
+      return true;
+    } else {
+      upgrade_rejected = true;
+      // Fall through to the default filter chain. The function calling this
+      // will send a local reply indicating that the upgrade failed.
```

Comment:
This means that this local reply will go through the encode filter chain for the default, right? SGTM if so.

Reply:
Correct.